### PR TITLE
Add the initial RFC template and instructions

### PR DESCRIPTION
**Pull request by tdcmeehan**
RFC about the RFC.  Please offer any feedback on this proposal (heavily modeled after PyTorch as credited).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## Step 1: Create an RFC
+RFCs are located in their own repository.
+
+To create one:
+
+1. Fork the https://github.com/prestodb/rfcs repository
+2. Copy the template file `RFC-0000-template.md` to `RFC-00xx-your-feature.md` and fill it out with your proposal.
+   The template is a guideline, feel free to add sections as appropriate
+3. You may also have the template simply link to another editor, like a Google Docs file, but please ensure that the
```
Comment (pratyakshsharma) [inline CONTRIBUTING.md:24]: I am a bit skeptical about giving this option/instruction. Even if google doc file is public, there is no guarantee it will be available after, say 1 year or so.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+This page contains instructions on how to propose and implement feature changes to Presto.
+
+# The Request for Comments
+
+## Overview of the process
+
+```mermaid
+graph TD;
```
Comment (pratyakshsharma) [inline CONTRIBUTING.md:8]: Curious to know if this `graph TD;` helps in some formatting/rendering? Why is this needed?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+   document is publicly visible.  This can make the template easier to add edit, but commenting doesn’t scale very well, so
+   please use this option with caution.
+
+## Step 2: Get Feedback on the RFC
+1. Submit a pull request [to the main repository](https://github.com/prestodb/rfcs) from your fork titled 
+`RFC-00xx-your-feature.md`
+2. Before your PR is ready for review,
+   [mark the PR as draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft).
+3. Once it’s ready for review, remove the draft label.
+4. In the description, include a short summary of your feature and a link to your RFC PR
```
Comment (pratyakshsharma) [inline CONTRIBUTING.md:34]: Small clarification: "In the description" -> which description are we referring to? Description of the PR raised?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+6. Build consensus. Those committers will review your PR and offer feedback. Revise your proposal as needed until
+   everyone agrees on a path forward. Additional forums you can share the proposal on include the
+   [the mailing list](https://lists.prestodb.io/g/presto-dev), and the
+   [Slack channel in #dev](https://communityinviter.com/apps/prestodb/prestodb). Tagging interested stakeholders
+   (identifiable via [CODEOWNERS](https://github.com/prestodb/presto/blob/master/CODEOWNERS)) can help with consensus building.
+
+_(Note: A proposal may get rejected if it comes with unresolvable drawbacks or if it’s against the long term plans of the Presto committers)_
+
+## Step 3: Implement your Feature
+1. If your RFC PR is accepted, the PR will be merged and markdown that adds the RFC will be merged it into the
```
Comment (pratyakshsharma) [inline CONTRIBUTING.md:45]: I think this statement can be simplified to "If your RFC is accepted, the PR will be merged."

File: README.md [hunk 5]
```diff
@@ -1,2 +1,27 @@
...
+usable.
+
+Smaller changes, including bug fixes and documentation improvements can be implemented and reviewed via the normal
+GitHub pull request workflow on the main Presto repo.
+
+RFCs are more suitable for design proposals that are too large to discuss on a feature-request issue, like adding new
+functionality to Presto, or if a discussion about the tradeoffs involved in a new addition are non-trivial.
+
+If you are unsure whether something should be an RFC or a feature-request issue, you can ask by opening an issue in the
+main Presto/Presto repository.
```
Comment (pratyakshsharma) [inline README.md:23]: nit: PrestoDB/Presto

File: RFC-0000-template.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+4. Code flow using bullet points or pseudo code as applicable
+5. Any new user facing metrics that can be shown on CLI or UI.
+
+## [Optional] Metrics
+
+How can we measure the impact of this feature?
+
+## [Optional] Other Approaches Considered
+
+(Based on the discussion, this may need to be updated with feedback from reviewers.)
```
Comment (pratyakshsharma) [inline RFC-0000-template.md:44]: nit: Let us remove the parentheses from start and end of this sentence?

**Review by pratyakshsharma — CHANGES_REQUESTED** [review]
Thank you for kick-starting this much awaited RFC process @tdcmeehan 

Few comments/clarifications from my side.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:8]: It makes the graph return **T**op **D**own.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:24]: We can say that before merge, it must be put into the main MD file.  This is just to reduce process and overhead to get the initial design out, but I can remove it.

File: CONTRIBUTING.md (hunk 3 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:34]: I'll remove it, it shouldn't be here

File: CONTRIBUTING.md (hunk 4 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:45]: Sure

**Comment by tdcmeehan** [comment]
Thanks for the feedback @pratyakshsharma, updated!

File: CONTRIBUTING.md (hunk 1 above)
Comment (pratyakshsharma) [reply CONTRIBUTING.md:24]: Yes, before merge it must be put into the main MD file. Then no issues.

Comment (pratyakshsharma) [reply CONTRIBUTING.md:24]: But I am bit leaning towards removing it, because ultimately the writeup has to be there in MD file only. We can wait to hear from others as well.

**Review by pratyakshsharma — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+review after the RFC has been accepted.
+
+If you are interested in working on the implementation for an accepted RFC, but
+cannot determine if someone else is already working on it, feel free to ask
+(e.g. by leaving a comment on the associated issue).
+
+
+## RFC Rejection
+Some RFC pull requests will be closed without the RFC being merged (as part of the rejection process). A closed RFC
+is closed because we want neither to think about evaluating the proposal nor about implementing the described feature
```
Comment (kiersten-stokes) [inline CONTRIBUTING.md:67]: I feel like this sentence is a bit confusing. I'm not sure my suggestion is any better, though 🙂 

```suggestion
is closed because we would prefer not to evaluate the proposal or implement the described feature
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+3. You may also have the template simply link to another editor, like a Google Docs file, but please ensure that the
+   document is publicly visible.  This can make the template easier to add edit, but commenting doesn’t scale very well, so
+   please use this option with caution.
+
+## Step 2: Get Feedback on the RFC
+1. Submit a pull request [to the main repository](https://github.com/prestodb/rfcs) from your fork titled 
+`RFC-00xx-your-feature.md`
+2. Before your PR is ready for review,
+   [mark the PR as draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft).
+3. Once it’s ready for review, remove the draft label.
```
Comment (kiersten-stokes) [inline CONTRIBUTING.md:33]: We may want to rephrase this for clarity since it seems like we won't be using traditional GH labels for our process

```suggestion
3. Once it’s ready for review, [move the PR out of draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#marking-a-pull-request-as-ready-for-review).
```

**Review by kiersten-stokes — COMMENTED** [review]
All makes sense to me! Just had 2 NITs

File: CONTRIBUTING.md (hunk 8 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:33]: Thanks!

**Comment by tdcmeehan** [comment]
Thanks @kiersten-stokes, I updated the docs!

**Review by kiersten-stokes — APPROVED** [review]

**Review by jaystarshot — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (aditi-pandit) [reply CONTRIBUTING.md:24]: I too prefer using the md file and PR for comments. It will provide consistency.

File: RFC-0000-template.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+Brief description of any existing issues, user requests or feature comparison with competitors which explains why the proposed feature might be needed. How the proposed feature helps our users? Explain the impact and value of this feature.
+
+### [Optional] Goals
+
+### [Optional] Non-goals
+
+## Proposed Implementation
+
+How do you intend to implement the feature? This section can be as detailed as possible with large subsections of its own, or may be a few sentences depending on the scope of the feature proposed. Explain the design in enough detail for existing users/contributors to understand. Design should include all the corner cases you can think of. Feel free to include any new SPI method signatures, class hierarchies or system contracts here. It is recommended to mention any methods, variables, classes which you think are needed to provide a broader view of the code changes. Please mention/describe the below on a high level -
```
Comment (aditi-pandit) [inline RFC-0000-template.md:30]: If the feature introduces SQL language changes or adds new system configuration, then we should describe these in this section. Right now these are in the adoption plan below. But I feel these user-interface changes should be described in the "Proposed implementation" as well. There might be feedback in terms of changing names or if they conflict with other better known usages of the new SQL etc.
